### PR TITLE
 Deprecated the selectors/actions and components that moved to the block editor module

### PR DESCRIPTION
**Pull request by youknowriad**
**Testing instructions**

- Check that if you call, `wp.data.select( 'core/editor' ).getSelectedBlock()` in the console it works but shows a warning.

- Change an `import RichText` and use `editor` instead of `block-editor` in one of the core blocks and check that there are warnings on the console when used.

### CODE REVIEW

**Comment by aduth** [comment]
This looks like a great way to keep ourselves honest and correct in code 👍 

There's still one missing. From the failing build:

```
wp.editor.withColors is deprecated and will be removed. Please use wp.blockEditor.withColors instead.
```

**Comment by youknowriad** [comment]
@aduth I think this comes from a plugin :p I wonder if it's time to remove the "plugins" test :)

**Comment by aduth** [comment]
Alternatively, we may consider another exception in the [console trapping](https://github.com/WordPress/gutenberg/blob/f1bbc004734ea157dd72ed5bf1f808c10efc0e84/packages/e2e-tests/config/setup-test-framework.js#L105) to permit warnings (and optionally errors, though probably not to start) from plugins, determined based on [ConsoleWarning#location](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#consolemessagelocation) (though, as I write this, the location is probably the deprecated script, not the plugin).

File: packages/editor/src/components/deprecated.js [hunk 1]
```diff
@@ -2,118 +2,139 @@
...
-	SkipToSelectedBlock,
-	URLInput,
-	URLInputButton,
-	URLPopover,
-	Warning,
-	WritingFlow,
-	withColorContext,
-	withColors,
-	withFontSizes,
+	Autocomplete as RootAutocompete,
```
Comment (talldan) [inline packages/editor/src/components/deprecated.js]:
```suggestion
	Autocomplete as RootAutocomplete,
```

File: packages/editor/src/components/deprecated.js [hunk 2]
```diff
@@ -2,118 +2,139 @@
...
+	return ( ...args ) => {
+		deprecated( 'wp.editor.' + name, {
+			alternative: 'wp.blockEditor.' + name,
+		} );
+
+		return func( ...args );
+	};
+}
+
+export const Autocomplete = deprecateComponent( 'Autocomplete', RootAutocompete );
```
Comment (talldan) [inline packages/editor/src/components/deprecated.js]:
```suggestion
export const Autocomplete = deprecateComponent( 'Autocomplete', RootAutocomplete );
```

**Comment by youknowriad** [comment]
What do you all think about removing the plugins tests? cc @gziolo How can we unblock this?

**Comment by gziolo** [comment]
We can either remove integration with 3rd party plugins or silence this warning on Travis. I would prefer the first one.

**Review by gziolo — APPROVED** [review]
I double checked all deprecated components are still exposed to avoid issues with 3rd party plugins.

I think we should still include an entry in the changelog about all those functions and components which will now warn on JS console so folks using npm package could take same actions.

Otherwise, everything works and looks good.

**Comment by youknowriad** [comment]
I'm hesitant about the docs as we don't have a formal process for Core deprecations yet. A document is indeed needed but probably not the one we have right now for plugin deprecations and also there's already a bunch of other Core deprecations.

**Comment by gziolo** [comment]
I was thinking about something like:

in `packages/editor/CHANGELOG.md`:
```md
### Deprecations

- It's now recommended to use the following components from `@wordpress/block-editor` package:
  - `Component1`
  - `Component2`
```

**Pull request merged**